### PR TITLE
use explicit manifest

### PR DESCRIPTION
**Pull request by LanieOkorodudu**
**Description of what you did:**
This PR updates the test to explicitly specify manifest for different test scenarios instead of relying entirely on the default manifest.
**Changes made**
- Existing UI-related test use the default manifest loaded from `BASE_URL`, since features such as gallery/two-up view are only available there.
- Viewer control and panel tests now use the IIIF Cookbook manifest: https://iiif.io/api/cookbook/recipe/0031-bound-multivolume/manifest.json
- Added a PDF manifest test using the Villanova multi-page PDF manifest: https://digital.library.villanova.edu/Item/vudl:294631/Manifest

The PDF test currently verifies that:
-  the PDF manifest URL is correctly loaded into the viewer
- Universal viewer initializes succesdsfully
- no visible loading errors are shown
- It is intentionally limited to manifest loading verification because the PDF page navigation does not currently update the viewer URL/canvas state in a reliable way for assertions.
I also increase the `jest.setTimeout` to 60000 to accommodate manifest loading that may take more time during PDF and cookbook manifest tests.

### CODE REVIEW

File: __tests__/test.js [hunk 1]
```diff
@@ -3,6 +3,20 @@ test.skip("Configuration options", () => {});
 const puppeteer = require("puppeteer");
 const { BASE_URL } = require("../scripts/testBaseUrl");
 
+// Cookbook manifest for viewer control tests
+const COOKBOOK_MANIFEST =
```
Comment (demiankatz) [inline __tests__/test.js]: Maybe we should give this a more specific name, since we'll likely have more cookbook manifests in the future -- perhaps `COOKBOOK_BOUND_MULTIVOLUME_MANIFEST`?

File: __tests__/test.js [hunk 2]
```diff
@@ -3,6 +3,20 @@ test.skip("Configuration options", () => {});
...
+  "https://iiif.io/api/cookbook/recipe/0031-bound-multivolume/manifest.json";
+
+// PDF manifest for PDF-specific behaviour
+const PDF_MANIFEST =
```
Comment (demiankatz) [inline __tests__/test.js]: Maybe this should be `PDF_MULTI_FILE_MANIFEST` -- since the edge case this manifest was provided to test is a set of multiple PDF files in a single manifest. One of the things we should confirm when loading the manifest is that the left sidebar appears with multiple files listed. (If we load a single-PDF manifest, the left sidebar should NOT appear).

File: __tests__/test.js [hunk 3]
```diff
@@ -3,6 +3,20 @@ test.skip("Configuration options", () => {});
...
+  "https://digital.library.villanova.edu/Item/vudl:294631/Manifest";
+
+const viewerUrl = (manifestUrl) => {
+  const separator = BASE_URL.includes("?") ? "&" : "?";
+
+  return `${BASE_URL}${separator}manifest=${encodeURIComponent(manifestUrl)}`;
```
Comment (demiankatz) [inline __tests__/test.js]: Does this really work? I thought the manifest argument had to be part of the hash parameters, not part of the base query parameters... in other words, `https://www.universalviewer.dev/?manifest=https://digital.library.villanova.edu/Item/vudl:294631` does not work, but `https://www.universalviewer.dev/#?manifest=https://digital.library.villanova.edu/Item/vudl:294631` does work. Note the `#?` instead of just `?`.

**Review by demiankatz — CHANGES_REQUESTED** [review]
Thanks, @LanieOkorodudu -- see below for some questions and suggestions.

File: __tests__/test.js (hunk 3 above)
Comment (LanieOkorodudu) [reply __tests__/test.js]: It did appear to work when I tested yesterday, but you are right that the manifest parameter should probably be part of the hash parameters rather than the base query parameters. I’ll update `viewerUrl` to use `#?manifest=` instead. Thanks

File: __tests__/test.js [hunk 4]
```diff
@@ -394,4 +414,40 @@ describe("Universal Viewer", () => {
...
+        waitUntil: "domcontentloaded",
+      });
+    });
+
+    it("loads PDF manifest successfully", async () => {
+      expect(page.url()).toContain(encodeURIComponent(PDF_MULTI_FILE_MANIFEST));
+
+      await page.waitForSelector(".uv", { visible: true });
+
+      const pageText = await page.evaluate(() => document.body.innerText);
```
Comment (demiankatz) [inline __tests__/test.js]: Is it perhaps also worth confirming that the `div.pdfContainer canvas` exists, since that's a sign that pdf.js has drawn something?

File: jest.setup.js [hunk 5]
```diff
@@ -1 +1 @@
-jest.setTimeout(10000);
\ No newline at end of file
+jest.setTimeout(60000);
```
Comment (demiankatz) [inline jest.setup.js]: Was it really necessary to raise the timeout this high? (If so, I don't mind -- but just making sure this wasn't a temporary change that got committed by accident).

**Review by demiankatz — CHANGES_REQUESTED** [review]
Thanks, @LanieOkorodudu, looking good -- just a couple of questions left!

File: jest.setup.js (hunk 5 above)
Comment (LanieOkorodudu) [reply jest.setup.js]: I increased it because some of the Puppeteer viewer-control tests were timing out occasionally in CI even though they were passing locally. I set it a bit higher to avoid flaky CI failures and keep the timeout handling centralized rather than adding per-test overrides everywhere.

Happy to lower it if you think 60000 is too high though.

Comment (demiankatz) [reply jest.setup.js]: A full minute is a pretty long time. Is it worth trying 30 seconds instead?

File: __tests__/test.js (hunk 4 above)
Comment (LanieOkorodudu) [reply __tests__/test.js]: I did try adding assertions to confirm that `div.pdfContainer canvas` exists, but the element was not consistently available in CI even though it rendered correctly locally. I tested it using the following assertions:

await page.waitForFunction(() => {
        return !!document.querySelector(
          "div.pdfContainer canvas"
        );
      });

        const canvasInfo = await page.evaluate(() => {
          const canvas = document.querySelector(
            "div.pdfContainer canvas"
          );

          if (!canvas) return null;
          return {
            width: canvas.width,
            height: canvas.height,
          };
        });
        expect(canvasInfo).not.toBeNull();
        expect(canvasInfo.width).toBeGreaterThan(0);
        expect(canvasInfo.height).toBeGreaterThan(0);
        
 I then added some logging to confirm that the manifest itself was loading and whether `div.pdfContainer canvas` existed after loading, and this is what the logs showed:

`current page url: http://localhost:4444/#?manifest=https%3A%2F%2Fdigital.library.villanova.edu%2FItem%2Fvudl%3A294631%2FManifest`

`div.pdfContainer canvas exists: false`

If there’s a better or more reliable way to assert/check the rendered PDF canvas in CI, I’d be happy to update the test.

File: jest.setup.js (hunk 5 above)
Comment (demiankatz) [inline jest.setup.js]: I see you thumbs-upped my suggestion to try 30 seconds, but this hasn't changed yet. Here's a convenient button to try applying the change if you still want to:

```suggestion
jest.setTimeout(30000);
```

(But if you tried it already and it didn't work, feel free to disregard this).

File: __tests__/test.js (hunk 4 above)
Comment (demiankatz) [reply __tests__/test.js]: Hmm, it's mysterious that the canvas isn't rendering in the test environment. Is it somehow possible that PDFObject mode is getting triggered instead of PDF.js mode? If the setting is getting set to use the alternate PDF library, that would certainly result in different HTML output... but I can't think why that would be the case.

**Review by demiankatz — CHANGES_REQUESTED** [review]
See below for some further thoughts... but if you'd like to get this one merged so you can move on to another set of tests, I don't mind moving forward even if we haven't fully solved the PDF mystery yet. We can refine that in the next round if necessary.

**Comment by LanieOkorodudu** [comment]
@demiankatz, I also noticed that targeting ".pdfContainer canvas" was unreliable in the test environment, the selector would time out even though the PDF was visibly rendering. Switching to a more generic "canvas" selector made the test pass consistently. My guess is that the container structure/classes are slightly different or not fully initialized yet in CI, while the canvas itself is still rendered correctly.

The updated test now waits for the Universal Viewer and iframe setup to finish, finds the active viewer frame, verifies that a canvas is rendered with valid dimensions, and checks that no visible loading errors appear.

**Review by demiankatz — APPROVED** [review]
Looks good to me -- thanks, @LanieOkorodudu!

**Comment by LanieOkorodudu** [comment]
> Looks good to me -- thanks, @LanieOkorodudu!

Thank you @demiankatz  for taking the time to look into this. I’m always grateful for your feedback. I  will merge it now.
